### ISSUE TITLE
perf(vulkan): keep top-k top-p temperature and penalties on the GPU

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12402]
Parent campaign: #11572. Frozen comparison ladder: #12093. Physical execution gate: #12098.

Order: **19 of 20** — post-spine follow-on, ordered by expected leverage and prerequisite structure.
Study receipt: `study_fea6c1bc31f94fcee01ec566289e7ca509c28e7a982068dbde17b36e33b333e3`. Evidence observed: 09/08/2026 13:45:00.

## For / Problem / Today / Better because / Witness
- **For:** fak-native Qwen3.8 on this machine's Ryzen AI Max+ 395 / Radeon 8060S (gfx1151, 64 GiB unified memory).
- **Problem:** Non-greedy serving reads the full vocabulary logits to the host even though the resident path can reduce greedy argmax on-device.
- **Today:** #12093 remains `NO_COMPARABLE_LOCAL_RESULT`; the sealed candidate/reference path is #12364 → #12365 → #12366 → #12367 → #12368 → #12369.
- **Better because:** Implement deterministic device-side penalties, temperature, top-k/top-p selection, and bounded candidate readback with an exact seeded reference contract.
- **Witness:** Fixed-seed token sequences match the CPU oracle, full-logit D2H copies are zero, result transfer is bounded, and sampled product-path c1 improves >=5% or the path stays off.

## Comparison and proof contract
The contextual dense Qwen3.8-27B Q4_K_M bars are 16.65 / 24.13 / 29.42 accepted output tok/s at c1/c4/c8; the 30.56–36.04 q38rocm rows remain challenge/watch evidence until matched. These numbers are context only. This issue cannot independently claim that fak exceeds another engine.

Physical promotion must use the same-commit sealed candidate/reference path above, exact model artifact and tensor inventory, prompt/output lengths, accepted-token denominator, driver/build/power identity, at least five interleaved repetitions, quality non-inferiority, and `fallback_count=0`. A microbenchmark can complete the software leaf but cannot unlock a product or comparison win.

## Field-borrow evidence
- **Pinned source:** https://github.com/ggml-org/llama.cpp/pull/27466
- **What transfers:** The merged upstream work demonstrates GPU-native long-row selection; fak adapts bounded selection to its own sampling semantics and quality witness.
- **Repo self-query:** `fak capabilities --json 'Vulkan GPU sampling top-k top-p'` → ABSENT after inspecting unrelated self-query results; raw Vulkan operations expose argmax only.
- **Exact fak seam:** `internal/model/hal.go:901-940; internal/compute/vulkan_ops.go:305-310`.

## Dependencies and scope fences
- **Coordinates with:** #12222, #9943, #12392 (R09).
- **Promotion requires:** #12222, #9943, #12392 (R09).
- **Pickup rule:** Open coordination and promotion edges do not block a bounded software or test slice. Use `Start blocked by: #N` only with source-bound proof that no executable disjoint slice remains.
- Native execution stays fak-owned. No silent llama.cpp or CPU fallback; llama.cpp is comparator-only.
- Preserve the exact Qwen3.8 artifact, prompt/output contract, accepted-token accounting, and quality gate.
- Scope is limited to: `internal/model/hal.go, internal/compute/vulkan_ops.go, internal/model/hal_vulkan_test.go`. Adjacent scheduler, allocator, codec, or model-format work is excluded unless named above.

## Done condition
- Implement deterministic device-side penalties, temperature, top-k/top-p selection, and bounded candidate readback with an exact seeded reference contract.

## Deterministic witness
`go test -tags vulkan ./internal/model ./internal/compute -run TestVulkanQwen38SamplingParityStrix -count=1`

## Physical promotion gate
- Fixed-seed token sequences match the CPU oracle, full-logit D2H copies are zero, result transfer is bounded, and sampled product-path c1 improves >=5% or the path stays off.
- Retain or default the candidate only when its source-bound physical effect clears the run's declared noise bound (normally ≥5%); otherwise record the elimination and advance to the next ranked lever.

```routing
lane: strix-vulkan-sampling
max_files: 3
max_steps: 7
class: post-spine follow-on
```
